### PR TITLE
Fix Li tag when switching the list style (issue #1307)

### LINKED ISSUE #1307: Missing li tags when switching list style
**Describe the bug**
The Aztec editor does not add li tags when changing the list style of an existing list

**To Reproduce**
1. Open demo App
2. Click on 'Empy Demo'
3. Enter 3 lines of an unordered list
4. select middle line and change it to ordered list
5. switch to code editor view

**Expected behavior**
```
<ul>
  <li>line 1</li>
</ul>
<ol>
  <li>line 2</li>
</ol>
<ul>
  <li>line 3</li>
</ul>
```

**Actual behavior**
```
<ul>
  <li>line 1</li>
</ul>
<ol>line 2</ol>
<ul>
  <li>line 3</li>
</ul>
```

**Smartphone (please complete the following information):**
 - Device:  iPhone6S
 - OS: iOS14.1

### PR DESCRIPTION
**Pull request by elangbamjohnson**
Fixes #1307 

To test:
Please tests with the steps provided in issue #1307

### CODE REVIEW

**Comment by elangbamjohnson** [comment]
Can someone please review this PR?

**Comment by hypest** [comment]
EDIT: managed to replicate the original issue so, ignore my comment below.

Left a [related comment in the issue itself](https://github.com/wordpress-mobile/AztecEditor-iOS/issues/1307#issuecomment-996647771), please have a look 🙇‍♂️

**Comment by hypest** [comment]
Thanks for the PR @elangbamjohnson!

@tanner, can you take a look when you get a chance? And, friendly reminder to check if this proposal affects Aztec's behavior in the block-editor or not. Thanks!

**Comment by elangbamjohnson** [comment]
> Thanks for the PR @elangbamjohnson!
> 
> @Tanner, can you take a look when you get a chance? And, friendly reminder to check if this proposal affects Aztec's behavior in the block-editor or not. Thanks!

I'll really appreciate if this can review sooner than later, I desperately need this fix in one of my project :) thanks for understanding

**Comment by twstokes** [comment]
> > Thanks for the PR @elangbamjohnson!
> > @Tanner, can you take a look when you get a chance? And, friendly reminder to check if this proposal affects Aztec's behavior in the block-editor or not. Thanks!
> 
> I'll really appreciate if this can review sooner than later, I desperately need this fix in one of my project :) thanks for understanding

Apologies for the delay @elangbamjohnson - I've been made aware of this PR and will look at it today. 🙇

File: Aztec/Classes/TextKit/TextView.swift [hunk 1]
```diff
@@ -1105,8 +1105,12 @@ open class TextView: UITextView {
         toggle(formatter: formatter, atRange: range)
 
         let liFormatter = LiFormatter(placeholderAttributes: typingAttributes)
-        toggle(formatter: liFormatter, atRange: range)
+        let isOlTagPresent = formatter.present(in: storage, at: range)
+        let isLiTagPresent = liFormatter.present(in: storage, at: range)
 
+        if (!isOlTagPresent && isLiTagPresent) || (isOlTagPresent && !isLiTagPresent) {
```
Comment (twstokes) [inline Aztec/Classes/TextKit/TextView.swift]: Could this be refactored to a smaller exclusive OR?

e.g. `if (isOlTagPresent != isLiTagPresent) {`

File: Aztec/Classes/TextKit/TextView.swift [hunk 2]
```diff
@@ -1122,8 +1126,12 @@ open class TextView: UITextView {
         toggle(formatter: formatter, atRange: range)
 
         let liFormatter = LiFormatter(placeholderAttributes: typingAttributes)
-        toggle(formatter: liFormatter, atRange: range)
-        
+        let isOlTagPresent = formatter.present(in: storage, at: range)
+        let isLiTagPresent = liFormatter.present(in: storage, at: range)
+
+        if (!isOlTagPresent && isLiTagPresent) || (isOlTagPresent && !isLiTagPresent) {
```
Comment (twstokes) [inline Aztec/Classes/TextKit/TextView.swift]: Could this be refactored to a smaller exclusive OR?

e.g. `if (isOlTagPresent != isLiTagPresent) {`

Comment (elangbamjohnson) [reply Aztec/Classes/TextKit/TextView.swift]: Good one! Make the change

File: Aztec/Classes/TextKit/TextView.swift (hunk 1 above)
Comment (elangbamjohnson) [reply Aztec/Classes/TextKit/TextView.swift]: Make the change

Comment (twstokes) [reply Aztec/Classes/TextKit/TextView.swift]: Thanks! And good catch on dropping the parenthesis.

File: Aztec/Classes/TextKit/TextView.swift (hunk 2 above)
Comment (twstokes) [reply Aztec/Classes/TextKit/TextView.swift]: Thanks! And good catch on dropping the parenthesis.

**Review by twstokes — APPROVED** [review]
Thank you for contributing @elangbamjohnson! I've tested these changes and they look good to me.

- Existing unit tests passed ✅ 
- Testing steps in https://github.com/wordpress-mobile/AztecEditor-iOS/issues/1307 passed ✅ 
- Smoke tests ✅ 

These will be merged to `develop` for inclusion on the next release.

**Pull request merged**